### ISSUE TITLE
FUTDC checking files in NuGet cache

### ISSUE THREAD

**Issue by drewnoakes** [dotnet/project-system#8805]
The FUTDC is supposed to filter out inputs from certain "non-modifiable" locations on disk. This helps keep the check performant.

Building this repo with verbose logging shows:

```
3>FastUpToDate:     Adding AdditionalFiles inputs: (Microsoft.VisualStudio.Editors)
3>FastUpToDate:         D:\repos\project-system\src\Microsoft.VisualStudio.Editors\PublicAPI.Unshipped.txt (Microsoft.VisualStudio.Editors)
3>FastUpToDate:         C:\Users\drnoakes\.nuget\packages\microsoft.visualstudio.sdk.analyzers\16.10.10\build\AdditionalFiles\vs-threading.MainThreadSwitchingMethods.txt (Microsoft.VisualStudio.Editors)
3>FastUpToDate:         C:\Users\drnoakes\.nuget\packages\microsoft.visualstudio.threading.analyzers\17.4.16-alpha\build\AdditionalFiles\vs-threading.MainThreadAssertingMethods.txt (Microsoft.VisualStudio.Editors)
3>FastUpToDate:         C:\Users\drnoakes\.nuget\packages\microsoft.visualstudio.sdk.analyzers\16.10.10\build\AdditionalFiles\BannedSymbols.txt (Microsoft.VisualStudio.Editors)
3>FastUpToDate:         C:\Users\drnoakes\.nuget\packages\microsoft.visualstudio.threading.analyzers\17.4.16-alpha\build\AdditionalFiles\vs-threading.MainThreadSwitchingMethods.txt (Microsoft.VisualStudio.Editors)
3>FastUpToDate:         C:\Users\drnoakes\.nuget\packages\microsoft.visualstudio.threading.analyzers\17.4.16-alpha\build\AdditionalFiles\vs-threading.LegacyThreadSwitchingMembers.txt (Microsoft.VisualStudio.Editors)
3>FastUpToDate:         C:\Users\drnoakes\.nuget\packages\microsoft.visualstudio.sdk.analyzers\16.10.10\build\AdditionalFiles\vs-threading.LegacyThreadSwitchingMembers.txt (Microsoft.VisualStudio.Editors)
3>FastUpToDate:         D:\repos\project-system\src\Microsoft.VisualStudio.Editors\PublicAPI.Shipped.txt (Microsoft.VisualStudio.Editors)
3>FastUpToDate:         C:\Users\drnoakes\.nuget\packages\microsoft.visualstudio.projectsystem\17.5.396-pre\build\AdditionalFiles\vs-threading.MainThreadAssertingMethods.txt (Microsoft.VisualStudio.Editors)
3>FastUpToDate:         C:\Users\drnoakes\.nuget\packages\microsoft.visualstudio.sdk.analyzers\16.10.10\build\AdditionalFiles\vs-threading.MembersRequiringMainThread.txt (Microsoft.VisualStudio.Editors)
3>FastUpToDate:         C:\Users\drnoakes\.nuget\packages\microsoft.visualstudio.projectsystem\17.5.396-pre\build\AdditionalFiles\vs-threading.MainThreadSwitchingMethods.txt (Microsoft.VisualStudio.Editors)
3>FastUpToDate:         C:\Users\drnoakes\.nuget\packages\microsoft.visualstudio.sdk.analyzers\16.10.10\build\AdditionalFiles\vs-threading.MainThreadAssertingMethods.txt (Microsoft.VisualStudio.Editors)
3>FastUpToDate:         C:\Users\drnoakes\.nuget\packages\microsoft.visualstudio.threading.analyzers\17.4.16-alpha\build\AdditionalFiles\vs-threading.MembersRequiringMainThread.txt (Microsoft.VisualStudio.Editors)
```

These `AdditionalFiles` files should not be being checked as they're located within the NuGet cache.

**Comment by drewnoakes** [comment]
Relevant code for investigation: https://github.com/dotnet/project-system/blob/7c1339a2012c23946e8c41257e0c52773e530c2e/src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/UpToDateCheckImplicitConfiguredInput.cs#L707-L710

**Comment by drewnoakes** [comment]
Another hit, when using the `Nullable` package, this time in `Compile` items:

```
1>FastUpToDate:     Adding Compile inputs:
1>FastUpToDate:         C:\Users\drnoakes\.nuget\packages\nullable\1.3.1\contentFiles\cs\net40\Nullable\NotNullIfNotNullAttribute.cs
```

**Comment by drewnoakes** [comment]
Turns out we weren't using the `ProjectFileClassifier` at all for input items. Fixed in #8831.
